pybaseball/cache/cache_config.py

```diff
@@ -0,0 +1,74 @@
+import inspect
+import logging
+import os
+import pathlib
+from datetime import timedelta
+from typing import Any, Optional, Text, cast
+
+# from typing_extensions import Literal
+import pandas as pd
+
+from . import file_utils
+
+# CacheType = Literal['csv', 'parquet']
+
+
+class CacheConfig:
+    DEFAULT_CACHE_DIR = os.path.join(pathlib.Path.home(), '.pybaseball', 'cache')
+    DEFAULT_EXPIRATION = 7  # number of days to cache by default
+    DEFAULT_CACHE_TYPE = 'parquet'
+    CFG_FILENAME = 'cache_config.json'
+
+    # Use this and __new__ to make this a singleton. Only one ever exists.
+    __INSTANCE__ = None
+    # pylint: disable=too-many-arguments
+
+    def __new__(cls, enabled: bool = False, cache_directory: str = None, default_expiration: int = None,
+                cache_type: Optional[str] = None) -> 'CacheConfig':
+        if not CacheConfig.__INSTANCE__:
+            CacheConfig.__INSTANCE__ = super(CacheConfig, cls).__new__(cls)
+
+        CacheConfig.__INSTANCE__._set(enabled, cache_directory, default_expiration, cache_type)
+        return CacheConfig.__INSTANCE__  # type: ignore
+
+    def _set(self, enabled: bool = False, cache_directory: str = None, default_expiration: int = None,
+             cache_type: Optional[str] = None) -> None:
```

Comment:
I think we should remove `cache_directory` from the API. This is on me really.

Reply:
I have a fallback idea. I'll add it and see what you think.

Reply:
Okay I removed cache_directory as a value in the new/set for this object. If the user wants to override I've added an environment variable.